botocore/operation.py

```diff
@@ -77,6 +80,30 @@ def call(self, endpoint, **kwargs):
                           model=self.model,
                           params=kwargs)
         request_dict = self.build_parameters(**kwargs)
+
+        service_name = self.service.service_name
+        service_model = self.session.get_service_model(service_name)
+        credentials = self.session.get_credentials()
+        scoped_config = self.session.get_scoped_config()
+        event_emitter = self.session.get_component('event_emitter')
+        resolver = self.session.get_component('endpoint_resolver')
+        scheme = endpoint.host.split(':')[0]
+        if endpoint.region_name is None:
```

Comment:
It's a copy of what's in clients, more of less. I'm on the fence about whether to try and make a utility method, because the client's version of this code is pretty clean right now.